bootstrap/bootstrap.go

```diff
@@ -91,9 +99,14 @@ func (b *Bootstrap) Run(ctx context.Context) (exitCode int) {
 		}
 	}()
 
+	span, ctx, stopper := b.startTracing(ctx)
+	defer stopper()
+	var err error
+	defer func() { tracetools.FinishWithError(span, err) }()
+
 	// Tear down the environment (and fire pre-exit hook) before we exit
 	defer func() {
-		if err := b.tearDown(); err != nil {
+		if err = b.tearDown(ctx); err != nil {
```

Comment:
Basic pattern here is to let the code within a function scope set errors, deal with it as fit, maybe returning soon. The defer will catch the error and properly set it on the span.